providers/implementations/ciphers/ciphercommon_gcm.c

```diff
@@ -48,6 +55,7 @@ static int gcm_init(void *vctx, const unsigned char *key, size_t keylen,
         return 0;
 
     ctx->enc = enc;
+    ctx->tls_enc_records = 0;
```

Comment:
Should it be moved to the `if (key != NULL)` block? IMO the counter should be reset only on key changes.

Reply:
Yes :(